plot/plot.app.js

```diff
@@ -1157,16 +1289,128 @@
                     if (newValue) vm.groups = newValue;
                 });
             }])
-            .directive('plot', ['$rootScope', '$timeout', function ($rootScope, $timeout) {
-
+            .directive('plot', ['$rootScope', '$timeout','ConfigUtils', function ($rootScope, $timeout,ConfigUtils) {
+                
                 return {
                     link: function (scope, element) {
                         scope.$watch('plots', function (plots) {
                             if (plots) {
                                 $timeout(function() {
+
                                     for (var i = 0; i < plots.length; i++) {
                                         if (plots[i].id == element[0].attributes['plot-id'].nodeValue) {
-                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config).then(function () {
+                                            Plotly.newPlot(element[0], plots[i].plot_values.data, plots[i].plot_values.layout, plots[i].plot_values.config)
+                                            .then(                                          
+                                                element[0].on('plotly_hover', function(data){
```

Comment:
seems like data will be the same in each of the events, `plotly_hover`, `plotly_unhover`, `plotly_click`, and `plotly_legendclick`. Let's put the full description of data for each plot type here instead.

Reply:
Done